src/options.rs

```diff
@@ -4,15 +4,166 @@
 //! operation may be completed.
 
 use core::{fmt, str::FromStr};
+use std::ops::Add;
 
 use crate::{
     components::{Date, ZonedDateTime},
-    TemporalError,
+    TemporalError, TemporalResult, NS_PER_DAY,
 };
 
 mod increment;
 pub use increment::RoundingIncrement;
 
+// ==== RoundingOptions / DifferenceSettings ====
+
+pub(crate) enum DifferenceOperation {
+    Until,
+    Since,
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct DifferenceSettings {
+    pub(crate) largest_unit: Option<TemporalUnit>,
+    pub(crate) smallest_unit: Option<TemporalUnit>,
+    pub(crate) rounding_mode: Option<TemporalRoundingMode>,
+    pub(crate) increment: Option<RoundingIncrement>,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct RoundingOptions {
+    pub(crate) largest_unit: Option<TemporalUnit>,
+    pub(crate) smallest_unit: Option<TemporalUnit>,
+    pub(crate) rounding_mode: Option<TemporalRoundingMode>,
+    pub(crate) increment: Option<RoundingIncrement>,
+}
+
+// Note: Specification does not clearly state a default, but
+// having both largest and smallest unit None would auto throw.
+
+impl Default for RoundingOptions {
+    fn default() -> Self {
+        Self {
+            largest_unit: Some(TemporalUnit::Auto),
+            smallest_unit: None,
+            rounding_mode: None,
+            increment: None,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct ResolvedRoundingOptions {
+    pub(crate) largest_unit: TemporalUnit,
+    pub(crate) smallest_unit: TemporalUnit,
+    pub(crate) increment: RoundingIncrement,
+    pub(crate) rounding_mode: TemporalRoundingMode,
+}
```

Comment:
Praise: Loving this new API! I think it really makes sense to have an options struct that resolves all the `Option`s in `RoundingOptions`, instead of having to `unwrap` everywhere.